cmake/CmakeFunctionHelper.cmake

```diff
@@ -100,7 +105,7 @@ set(LOCATION "${SYCLBLAS_GENERATED_SRC}/${blas_level}/${func}/")
 foreach(executor ${executor_list})
   foreach(data ${data_list})
     cpp_type(cpp_data ${data})
-    set(container_list "BufferIterator<${cpp_data},codeplay_policy>")
+    set(container_list "typename ${sycl_blas_policy}::template buffer_t<${cpp_data}>")
```

Comment:
This is causing an issue. Put it inside if usm and revert the buffer iterator in else